core/Plugin/Menu.php

```diff
@@ -30,16 +30,6 @@
  */
 class Menu
 {
-    protected $module = '';
```

Comment:
I think you are confusing "called in total" and "called in each plugin". It is called just a few times in each plugin, and since that method return the plugin name that wouldn't cache the 50 calls, only 1-2-3 calls which is why I'm saying it's useless.

The problem is that we want to leave the constructor free to use for subclasses to use dependency injection. If those classes need to call the parent constructor this is error prone as we can expect beginner plugin developers to miss that (and the errors will be hard to debug). Even for experienced developers this is just a pain for no gain.